src/FileCommandFacility.cpp

```diff
@@ -0,0 +1,262 @@
+        }
+        auto obj = arr[0];
+        arr.erase(0);
+        if (! obj.is_object()) {
+            std::string msg = "want: object, got: ";
+            msg += obj.dump(4); // fixme: temp for debugging!
+            throw StreamCorrupt(ERS_HERE, name, msg);
+        }
+        return obj;
+    }
+
+    virtual void put(object_t obj) {
+        arr.push_back(obj);
+    }
+
+    virtual void flush() {
+        if (arr.empty()) { return; }
+        w() << arr;
+        arr.empty();
+    }
+};
+
+
+// Note: this is registered as fileCommandFacility (lower-case "f"),
+// available on CLI as file:// scheme or simple filesystem path name.
+// Since files are not bi-directional, this will form an output
+// filename based on the input with "-out" appended to the base
+// filename.  The same extention and thus format assumptions used by
+// the input are kept for output.
+struct fileCommandFacility : public CommandFacility {
+
+    std::fstream istr;
+    std::unique_ptr<ObjectStream> ios;
+
+    virtual ~fileCommandFacility() {
+        // assure these die first.
+        ios.reset();
+    }
+    fileCommandFacility(const fileCommandFacility&) = delete;
+    fileCommandFacility(const fileCommandFacility&&) = delete;
+    fileCommandFacility& operator=(const fileCommandFacility&) = delete;
+    fileCommandFacility& operator=(const fileCommandFacility&&) = delete;
+
+    fileCommandFacility(std::string uri) : CommandFacility(uri) {
+
+        auto dot = uri.find_last_of(".");
+        auto sep = uri.find("://");
+        if (dot == std::string::npos) {
+            throw UnsupportedUri(ERS_HERE, uri);
+        }
+
+        std::string ext = uri.substr(dot+1);
+        std::string scheme = "";
+        std::string iname = uri;
+
+        if (sep == std::string::npos) { // simple path
+            scheme = "file";
+            iname = uri;
+        } else {                  // with scheme
+            scheme = uri.substr(0, sep);
+            iname = uri.substr(sep+3);
+        }
+        std::string bname = iname.substr(0, dot);
+
+        ERS_INFO("open: scheme:" << scheme << " ext:" << ext);
+
+        
+        if (scheme.empty() || scheme == "file") {
+            istr.open(iname, std::ios_base::in);
+            if (ext == "json") {
+                ios.reset(new JsonArray(iname, istr, true));
+            }
+            if (ext == "jstream") {
+                ios.reset(new JsonStream(iname, istr));
+            }
+            return;
+        }
+        throw UnsupportedUri(ERS_HERE, uri);
+    }
+
+
+    object_t recv() const {
+        return ios->get();
+    }
+
+
+    void run(DAQModuleManager& manager) const {
+
+        while (true) {
+            object_t command;
+
+            try {
+                command = this->recv();
+            }
+            catch (const StreamExhausted& e) {
+                ERS_INFO("Command stream end");
+                break;
+            }
+
+            manager.execute(command);
```

Comment:
This needs a try and catch loop here. ModuleManager can throw and we are not catching it.